tests/test_http.py

```diff
@@ -515,6 +515,19 @@ def test_send_expect_header(self):
             config = HttpResourcePathConfig()
             self.assertTrue(config.send_expect_on_put)
 
+    def test_collect_memory_usage(self):
+        # Ensure environment variable LSST_HTTP_COLLECT_MEMORY_USAGE is
+        # inspected to initialize the HttpResourcePathConfig class.
+        with unittest.mock.patch.dict(os.environ, {}, clear=True):
+            importlib.reload(lsst.resources.http)
+            config = HttpResourcePathConfig()
+            self.assertFalse(config.collect_memory_usage)
+
+        with unittest.mock.patch.dict(os.environ, {"LSST_HTTP_COLLECT_MEMORY_USAGE": "true"}, clear=True):
```

Comment:
Another approach would be to take one of the other real tests, patch the environment and then run with `assertLogs("DEBUG")` and see if you get a memory usage message in the log. It depends on whether you want to test that this flag is set or if this flag has a real effect.

Reply:
I considered that the tests of `time_this` would test that memory usage is actually collected when required. My intention here is to test that you can actually configure that from the environment.